src/ui/public/courier/data_source/search_source.js

```diff
@@ -268,6 +269,9 @@ export function SearchSourceProvider(Promise, Private, config) {
         val = normalizeSortRequest(val, this.get('index'));
         addToBody();
         break;
+      case 'fields':
+        state[key] = [...state[key] || [], ...val];
```

Comment:
The `||` takes precedence over `...`, I added the parenthesis to make this more clear. Good call on deduping them!